### PR TITLE
Implement conversation (session) management in system agent

### PR DESCRIPTION
**Pull request by GeorgeNgMsft**
**Overview**
This PR adds natural-language conversation/session management capabilities across all three TypeAgent surfaces: the dispatcher (backend), the CLI, and the shell (Electron UI). Users can now say things like "create a new conversation called work" or "switch to my research session" and have those handled end-to-end.

**Dispatcher**
- Extended SessionAction schema with two new action types: switchSession and deleteSession
- Renamed showSessionInfo → showConversationInfo for consistency
- Rewrote executeSessionAction to use clientIO.takeAction("manage-conversation", payload) instead of calling processCommandNoLock directly — this decouples the dispatcher from CLI-specific command parsing and lets each client (CLI, shell) handle the action natively
- Enriched the system agent description and schema JSDoc comments with extensive examples to improve NL translation accuracy
- 

**CLI** 
- Added confirmYesNo() utility that reuses the active readline interface instead of creating a competing one — fixes a prior bug where the y/n prompt would fight with the main input loop
- Renamed internal question() → question_internal() to make the reuse surface explicit
- Added a manage-conversation action handler in createEnhancedClientIO that dispatches to handleConversationCommand
- Exposed getConversationCommandContext() so the IO handler can route actions to the existing conversation command logic
- Added conversation info subcommand to show current session name and ID
- Fixed a spinner visibility issue: spinner is now stopped before awaiting pendingChoicePromise

**Shell**
- Added a manage-conversation case in the renderer's client action switch that handles all six subcommands (new, list, info, switch, delete, rename) via the existing getClientAPI() session methods
- All user-visible strings are HTML-escaped via a new escapeHtml() helper

**Tests** 
- New Jest test suite covering all five action types, including edge cases (no name, names with spaces, unknown action)

**Docs**
- Updated agentServerSessions.md architecture doc
- Updated dispatcher.md with session management context

### CODE REVIEW

File: ts/packages/shell/src/renderer/src/main.ts [hunk 1]
```diff
@@ -373,6 +373,275 @@ function registerClient(
...
+                                        );
+                                        break;
+                                    }
+                                    const sessions = await api.sessionList();
+                                    const match = sessions.find(
+                                        (s) =>
+                                            s.name.toLowerCase() ===
+                                            payload.name!.toLowerCase(),
+                                    );
+                                    if (!match) {
```
[the remark this answers is not in the document]
Comment (GeorgeNgMsft) [reply ts/packages/shell/src/renderer/src/main.ts:493]: This was incorrectly described in the PR description

File: ts/packages/shell/src/renderer/src/main.ts [hunk 2]
```diff
@@ -373,6 +373,275 @@ function registerClient(
...
+                                                    "A name is required to create a new conversation.",
+                                                kind: "warning",
+                                            },
+                                            "session",
+                                            undefined,
+                                        );
+                                        break;
+                                    }
+                                    const created = await api.sessionCreate(
+                                        payload.name,
```
[the remark this answers is not in the document]
Comment (GeorgeNgMsft) [reply ts/packages/shell/src/renderer/src/main.ts:403]: This is fine for now, we'll leave it as a TODO
